dashboard/src/main/java/org/openedx/dashboard/presentation/DashboardFragment.kt

```diff
@@ -347,6 +349,7 @@ private fun CourseItem(
     windowSize: WindowSize,
     onClick: (EnrolledCourse) -> Unit
 ) {
+    val config = getKoin().get<Config>()
```

Comment:
why config is injected here, it need to be injected inside viewModel, and all config fields should be provided throw uiState into UI

Reply:
@k1rill I believe we can explore this option, but I’m curious about the potential benefits of taking the longer route. I haven’t come across any recommendation guidelines regarding this approach. Could you share more insights on what we aim to achieve through this extended path?

Reply:
@HamzaIsrar12 we are trying to build **Clean architecture** with **SOLID** principles. UI shouldn't work with logic directly. There is injection done inside UI, and even worse that it is done inside composable function. It will be executed on every recomposition. Also this config can't be changed runtime at this moment, so we should init it once on the ViewModel init, and do not search needed singleton in the DI graph on every layout change.